sklearn/metrics/tests/test_pairwise_distances_reduction.py

```diff
@@ -1226,3 +1344,36 @@ def test_sqeuclidean_row_norms(
     with pytest.raises(ValueError):
         X = np.asfortranarray(X)
         sqeuclidean_row_norms(X, num_threads=num_threads)
+
+
+def test_argkminclassmode_strategy_consistent():
+    rng = np.random.RandomState(1)
+    X = rng.rand(100, 10)
+    Y = rng.rand(100, 10)
+    k = 5
+    metric = "euclidean"
```

Comment:
I am confused why this works. I thought that `ArgKminClassMode` was not `usable_for` Euclidean based metrics. I suppose that this is using the non GEMM optimized, vector-wise implementation of Euclidean distances.

Reply:
I think this test should be updated to test with `manhattan` for now, and once the Euclidean specialization is implemented, let's test for both.

Reply:
I think I forgot to update this test, good catch. Fixed.